cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+	//make sure files meet the usage requirements of globalAlignment.go
+	faOne, faDoneOne := fasta.NextFasta(inputFileOne)
+	faTwo, faDoneTwo := fasta.NextFasta(inputFileTwo)
+	fasta.ToUpper(faOne) //raven added these 2 lines to prevent lowercase letters from indexing out of range in align/linearGap.go
+	fasta.ToUpper(faTwo)
+	if faDoneOne || faDoneTwo {
+		log.Fatalf("error, unable to read .fa files, check for > symbol before each name and that each fasta entry is only one line. Check to make sure there are only four bases or N, globalAlignment is unable to use anything outside A-T-G-C-N.\n")
+	}
+	numSeqOne := CountSeqIdx(inputFileOne)
+	numSeqTwo := CountSeqIdx(inputFileTwo)
+	if numSeqOne > 1 || numSeqTwo > 1 {
+		log.Fatalf("multiple sequnces detected in .fa files: %v sequences in the first .fa file and %v sequences in the second .fa file. This program is designed for .fa files with only 1 sequence in them\n", numSeqOne, numSeqTwo)
+	}
+	//attempt at Dan's len(faOne) suggestion
+	//if len(string(faOne)) != 1 || len(string(faTwo)) != 1 {
+	//log.Fatalf("multiple sequnces detected in .fa files. This program is designed for .fa files with only 1 sequence in them.\n")
+	//}
+
+	//needleman wunsch (global alignment)
+	//raven's note: scoring matrices are in align/align.go, e.g. HumanChimp matrix, same as what UCSC Genome Browser uses
+	//raven's note: ConstGap is a function in align/linearGap.go, with only constant gapPen (gap penalty)
+	//bestScore, aln := align.ConstGap(faOne.Seq, faTwo.Seq, align.HumanChimpTwoScoreMatrix, -430)
+	//fmt.Printf("Alignment score is %d, cigar is %v \n", bestScore, aln)
+	//raven's edit: try AffineGap in align/affineGap.go, with diffrent gapOpen and gapExtend penalty, since gapExtend<gapOpen, encourages big gap in order to align better in unbroken chunks
+	bestScore, aln := align.AffineGap(faOne.Seq, faTwo.Seq, align.HumanChimpTwoScoreMatrix, -600, -150)
+	fmt.Printf("Using AffineGap, Alignment score is %d, cigar is %v \n", bestScore, aln)
+
+	//TODO: raven's cigarToBed main body starts here
+	//type definitions: Cigar is defined in align/align.go; aln is a []cigar, according to ConstGap defined in align/linearGap.go
+	//modeled after: mafIndels.go, wigPeaks.go
+	//insertion bed
+	insBed := fileio.EasyCreate(outIns_bed)
+	defer insBed.Close()
+	//TODO: now have to write FirstPos, but find a way to grab from fasta header
+	ChromCurrent := FirstPos_InsBed - 1 //initialize a variable to keep track of current position on chromosome
+	ChromStart := FirstPos_InsBed - 1   //initialize variables for ChromStart and ChromEnd, will need to set them equal to one another during updates, but may be more efficient to have fewer variables
+	ChromEnd := FirstPos_InsBed - 1
+	for i := 0; i < len(aln)-1; i++ { //loop through each entry in the []cigar, start from 0, end at len-2 so can still assess aln[i+1]
+		if aln[i].Op == 0 && aln[i+1].Op == 1 { //if there is an M before I, then write to bed
```

Comment:
You might consider using the Op names instead of the integer here too. Though because your code is well commented, it is less of an issue.